WordPress/Classes/Networking/SharingServiceRemote.swift

```diff
@@ -0,0 +1,297 @@
+
+            },
+            failure: { (operation: AFHTTPRequestOperation!, error: NSError!) -> Void in
+                failure?(error)
+            })
+    }
+
+
+    /**
+    *  @brief      Fetches the current user's list of keyring connections.
+    *
+    *  @param      success     An optional success block accepting an array of `KeyringConnection` objects.
+    *  @param      failure     An optional failure block accepting an `NSError` argument.
+    */
+    public func getKeyringConnections(success: ([KeyringConnection] -> Void)?, failure: (NSError! -> Void)?) {
+        let endpoint = "me/keyring-connections"
+        let path = self.pathForEndpoint(endpoint, withVersion: ServiceRemoteRESTApiVersion_1_1)
+
+        api.GET(path,
+            parameters: nil,
+            success: { (operation:AFHTTPRequestOperation!, response:AnyObject!) -> Void in
+                guard let onSuccess = success else {
+                    return
+                }
+
+                let responseDict = response as! NSDictionary
+                let connections:Array = responseDict.arrayForKey(ConnectionDictionaryKeys.connections)
+                let keyringConnections:[KeyringConnection] = connections.map { (let dict) -> KeyringConnection in
+                    let conn = KeyringConnection()
+                    // TODO: Need to see how this guy is formatted.
+                    // conn.additionalExternalUsers = dict.arrayForKey(ConnectionDictionaryKeys.additionalExternalUsers)
+                    conn.dateExpires = DateUtils.dateFromISOString(dict.stringForKey(ConnectionDictionaryKeys.expires))
+                    conn.dateIssued = DateUtils.dateFromISOString(dict.stringForKey(ConnectionDictionaryKeys.issued))
+                    conn.externalDisplay = dict.stringForKey(ConnectionDictionaryKeys.externalDisplay) ?? conn.externalDisplay
+                    conn.externalID = dict.stringForKey(ConnectionDictionaryKeys.externalID) ?? conn.externalID
+                    conn.externalName = dict.stringForKey(ConnectionDictionaryKeys.externalName) ?? conn.externalName
+                    conn.externalProfilePicture = dict.stringForKey(ConnectionDictionaryKeys.externalProfilePicture) ?? conn.externalProfilePicture
+                    conn.keyringID = dict.numberForKey(ConnectionDictionaryKeys.ID) ?? conn.keyringID
+                    conn.label = dict.stringForKey(ConnectionDictionaryKeys.label) ?? conn.label
+                    conn.refreshURL = dict.stringForKey(ConnectionDictionaryKeys.refreshURL) ?? conn.refreshURL
+                    conn.status = dict.stringForKey(ConnectionDictionaryKeys.status) ?? conn.status
+                    conn.service = dict.stringForKey(ConnectionDictionaryKeys.service) ?? conn.service
+                    conn.type = dict.stringForKey(ConnectionDictionaryKeys.type) ?? conn.type
+                    conn.userID = dict.numberForKey(ConnectionDictionaryKeys.userID) ?? conn.userID
```

Comment:
Very true, but I still think (architecturally speaking) remote object values should be populated by the Remote service that uses them.   The argument being, like Posts, different end points can return different flavors of the object, and the object's initializes shouldn't be responsible for that logic.  Of course that's not really something we need to worry about here (at least for now).